File: WebAPI/eFMS.API.SystemWeb/DocumentationManagement/eFMS.API.Documentation.DAL/Models/CsShipmentSurcharge.cs
using System;
using System.Collections.Generic;

namespace eFMS.API.Documentation.Service.Models
{
    public partial class CsShipmentSurcharge
    {
        public Guid Id { get; set; }
        public Guid Hblid { get; set; }
        public string Type { get; set; }
        public Guid ChargeId { get; set; }
        public decimal Quantity { get; set; }
        public short UnitId { get; set; }
        public decimal? UnitPrice { get; set; }
        public string CurrencyId { get; set; }
        public bool? IncludedVat { get; set; }
        public decimal? Vatrate { get; set; }
        public decimal Total { get; set; }
        public string PayerId { get; set; }
        public string ReceiverObject { get; set; }
        public string ReceiverId { get; set; }
        public string ObjectBePaid { get; set; }
        public string PaymentObjectId { get; set; }
        public bool? KickBack { get; set; }
        public DateTime? ExchangeDate { get; set; }
        public string Notes { get; set; }
        public string DocNo { get; set; }
        public string SettlementCode { get; set; }
        public string Csidsettlement { get; set; }
        public string CsstatusSettlement { get; set; }
        public DateTime? CsdateSettlement { get; set; }
        public string InvoiceNo { get; set; }
        public string PaymentRefNo { get; set; }
        public string AccountantId { get; set; }
        public DateTime? AccountantDate { get; set; }
        public string AccountantStatus { get; set; }
        public string AccountantNote { get; set; }
        public string ChiefAccountantId { get; set; }
        public DateTime? ChiefAccountantDate { get; set; }
        public string ChiefAccountantStatus { get; set; }
        public string ChiefAccountantNote { get; set; }
        public string Status { get; set; }
        public string SoaadjustmentRequestor { get; set; }
        public DateTime? SoaadjustmentRequestedDate { get; set; }
        public string SoaadjustmentReason { get; set; }
        public string UnlockedSoasaleMan { get; set; }
        public string UnlockedSoasaleManStatus { get; set; }
        public DateTime? UnlockedSoasaleManDate { get; set; }
        public string UnlockedSoadirector { get; set; }
        public string UnlockedSoadirectorStatus { get; set; }
        public DateTime? UnlockedSoadirectorDate { get; set; }
        public bool? Soaclosed { get; set; }
        public string Soano { get; set; }
        public string OtherSoa { get; set; }
        public string UserCreated { get; set; }
        public DateTime? DatetimeCreated { get; set; }
        public string UserModified { get; set; }
        public DateTime? DatetimeModified { get; set; }
    }
}

File: WebApp/src/app/shared/models/document/csShipmentSurcharge.ts
export class CsShipmentSurcharge {
    id: string = "00000000-0000-0000-0000-000000000000";
    hblid: string = null;
    type: string = null;
    chargeId: string = null;
    chargeNameEn: string = null;
    quantity: number = 1;
    unitId: number = null;
    unitPrice: number = null;
    currencyId: string = null;
    includedVat: boolean = null;
    vatrate: number = null;
    total: number = 0;
    payerId: string = null;
    objectBePaid: string = null;
    paymentObjectId: string = null;
    kickBack: boolean = null;
    exchangeDate: any = null;
    notes: string = null;
    settlementCode: string = null;
    csidsettlement: string = null;
    csstatusSettlement: string = null;
    csdateSettlement: any = null;
    invoiceNo: string = null;
    invoiceDate: any = null;
    seriesNo: string = null;
    paymentRefNo: string = null;
    accountantId: string = null;
    accountantDate: any = null;
    accountantStatus: string = null;
    accountantNote: string = null;
    chiefAccountantId: string = null;
    chiefAccountantDate: any = null;
    chiefAccountantStatus: string = null;
    chiefAccountantNote: string = null;
    status: string = null;
    userCreated: string = null;
    datetimeCreated: any = null;
    userModified: string = null;
    datetimeModified: any = null;
    exchangeRate: number = null;
    chargeCode: string = null;
    unit: string = null;
    hwbno: string = null;
    soaadjustmentReason: string = null;
    soaadjustmentRequestedDate: any = null;
    soaadjustmentRequestor: string = null;
    soaclosed: boolean = null;
    cdno: string = null;
    soano: string = null;
    currency: string = null;
    nameEn: string = null;
    otherSoa: string = null;
    partnerName: string = null;
    partnerShortName: string = null;
    payerName: string = null;
    payerShortName: string = null;
    receiverName: string = null;
    receiverShortName: string = null;
    unlockedSoadirector: string = null;
    unlockedSoadirectorDate: any = null;
    unlockedSoadirectorStatus: string = null;
    unlockedSoasaleMan: string = null;
    unlockedSoasaleManDate: any = null;
    unlockedSoasaleManStatus: string = null;
    isFromShipment: boolean = true;
    isRemaining: boolean = null;
    isSelected: boolean = null;
    isDeleted: boolean = null;
    voucherId: string = null;
    voucherIddate: string = null;
    voucherIdre: string = null;
    voucherIdredate: string = null;
    finalExchangeRate: string = null;
    acctManagementId: string = null;
    syncedFrom: string = null;
    paySyncFrom: string = null;

    quantityType: any = null;
    creditNo: string = null;
    debitNo: string = null;
    jobNo: string = null;
    mblno: string = null;
    hblno: string = null;
    advanceNo: string = null;

    debitCharge: string = null;
    paySoano: string = null;
    unitCode: string = null;
    chargeGroup: string = null;
    vatPartnerId: string = null;
    vatPartnerShortName: string = null;
    officeId: string = null;
    transactionType: string = null;

    // * Custom
    duplicateCharge: boolean = false;
    duplicateInvoice: boolean = false;
    isShowPartnerHeader: boolean = false;
    linkFee: boolean = null;
    linkChargeId: string = null;

    constructor(object?: any) {
        const self = this;
        for (const key in object) {
            if (self.hasOwnProperty(key.toString())) {
                self[key] = object[key];
            }
        }
    }
}
